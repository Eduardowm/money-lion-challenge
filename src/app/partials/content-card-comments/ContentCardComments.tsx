'use client'

import Button from '@/components/ui/button/Button'
import { Comment } from '@/data/entities/CommentEntity'
import * as Avatar from '@radix-ui/react-avatar'
import { useState } from 'react'

interface ContentCardCommentsProps {
  comments: Comment[]
}

function ContentCardComments({ comments }: ContentCardCommentsProps) {
  const [showComments, setShowComments] = useState<boolean>(false)

  if (!showComments) {
    return (
      <div className="px-1">
        <Button onClick={() => setShowComments(true)}>
          View {comments.length} comment{comments.length > 1 ? 's' : ''}
        </Button>
      </div>
    )
  }

  return (
    <div className="flex flex-col px-1 gap-4">
      <h3 className="text-white text-sm sm:text-base font-semibold">Comments</h3>
      {comments.map((comment: any) => (
        <div key={comment.text} className="flex flex-col">
          <div className="flex items-start flex-wrap sm:flex-nowrap gap-2">
            <Avatar.Root className="inline-flex h-[45px] w-[45px] select-none items-center justify-center overflow-hidden rounded-full align-middle">
              <Avatar.Image
                className="rounded-full object-cover"
                src={comment.image}
                alt={comment.name}
                width={45}
                height={45}
              />

              <Avatar.Fallback
                className="text-gray-900 flex h-full w-full items-center justify-center bg-white font-medium"
                delayMs={600}
              >
                {comment.name[1].toUpperCase()}
              </Avatar.Fallback>
            </Avatar.Root>

            <div className="flex flex-col grow">
              <span className="text-gray-500 text-sm sm:text-base">{comment.name}</span>
              <p className="text-gray-200 text-sm sm:text-base">{comment.text}</p>
            </div>

            <span className="text-xs text-gray-500 self-end w-full sm:w-fit">{comment.likes} likes</span>
          </div>
        </div>
      ))}
    </div>
  )
}
export default ContentCardComments
