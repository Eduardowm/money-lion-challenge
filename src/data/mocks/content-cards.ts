import { ContentCard } from '../entities/ContentCardEntity'

export const mockRawContentCard = {
  id: 'abc-def-ghi',
  imageUri: '/images/mock.jpg',
  textData: {
    title: 'A mock title',
    subTitle: 'A mock subtitle',
    body: 'A mock content with a lot of text to fill the space',
    author: { first: 'Mock', last: 'author' },
  },
  metadata: {
    priority: 1,
    publishDate: '2021-01-01',
  },
  comments: [
    {
      text: 'A mock comment',
      author: 'Commenter 1',
      profilePic: '/images/mock.jpg',
      likes: 20,
    },
    {
      text: 'The second mock comment',
      author: 'Commenter 2',
      profilePic: '/images/mock2.jpg',
      likes: 50,
    },
  ],
}

export const mockContentCards: ContentCard[] = [
  {
    id: 'abc-def-ghi',
    title: 'A mock title',
    subtitle: 'A mock subtitle',
    content: 'A mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'Mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Commenter 1',
        image: '/images/mock.jpg',
        text: 'A mock comment',
        likes: 20,
      },
      {
        name: 'Commenter 2',
        image: '/images/mock2.jpg',
        text: 'The second mock comment',
        likes: 50,
      },
    ],
  },
  {
    id: 'ihg-fed-cba',
    title: 'An alternative mock title',
    subtitle: 'An alternative mock subtitle',
    content: 'An alternative mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'An alternative mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Alternative commenter 1',
        image: '/images/mock.jpg',
        text: 'An alternative mock comment',
        likes: 20,
      },
    ],
  },
]
