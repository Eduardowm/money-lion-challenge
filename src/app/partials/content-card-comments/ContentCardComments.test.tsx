import { render } from '@testing-library/react'

import { mockContentCards } from '@/data/mocks/content-cards'
import ContentCardComments from './ContentCardComments'

describe('ContentCardComments', () => {
  it('renders as expected', () => {
    const { container } = render(<ContentCardComments comments={mockContentCards[0].comments} />)

    expect(container).toMatchSnapshot()
  })
})
