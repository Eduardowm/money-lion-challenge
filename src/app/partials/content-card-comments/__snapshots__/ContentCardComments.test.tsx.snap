// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCardComments renders as expected 1`] = `
<div>
  <div
    class="px-1"
  >
    <button
      class="text-gray-500 hover:text-gray-300"
    >
      View 
      2
       comment
      s
    </button>
  </div>
</div>
`;
